pkg/iac-providers/kustomize/v3/load-dir.go

```diff
@@ -24,28 +24,29 @@ func (k *KustomizeV3) LoadIacDir(absRootDir string) (output.AllResourceConfigs,
 
 	files, err := utils.FindFilesBySuffixInDir(absRootDir, KustomizeFileNames())
 	if err != nil {
+		err = errorKustomizeNotFound(err)
 		zap.S().Error("error while searching for iac files", zap.String("root dir", absRootDir), zap.Error(err))
 		return allResourcesConfig, err
 	}
 
 	if len(files) == 0 {
-		err = errors.New("could not find a kustomization.yaml/yml file in the directory")
+		err = errorKustomizeNotFound(errors.New("could not find a kustomization.yaml/yml file in the directory"))
 		zap.S().Error("error while searching for iac files", zap.String("root dir", absRootDir), zap.Error(err))
 		return allResourcesConfig, err
 	}
 
 	if len(files) > 1 {
-		err = errors.New("a directory cannot have more than 1 kustomization.yaml/yml file")
+		err = errorMultipleKustomizeFile(errors.New("a directory cannot have more than 1 kustomization.yaml/yml file"))
 		zap.S().Error("error while searching for iac files", zap.String("root dir", absRootDir), zap.Error(err))
 		return allResourcesConfig, err
 	}
 
 	kustomizeFileName := *files[0]
 	yamlkustomizeobj, err := utils.ReadYamlFile(filepath.Join(absRootDir, kustomizeFileName))
 
-	if len(yamlkustomizeobj) == 0 {
-		err = fmt.Errorf("unable to read any kustomization file in the directory : %v", err)
-		zap.S().Error("error while searching for iac files", zap.String("root dir", absRootDir), zap.Error(err))
+	if err != nil {
+		err = fmt.Errorf("unable to read the kustomization file in the directory : %v", err)
```

Comment:
Recommend to move the error definitions into a variable that can be referenced easily by unit tests